text/0000-rename-distributed-ctx.md

```diff
@@ -0,0 +1,59 @@
+# Rename distributed context into tags and add support for Baggage.
+
+**Status:** `proposed`
+
+Rename Distributed-context to a less confusing term and restrict its use for observability.
+ 
+ 
+## Motivation
+
+Context Propagation is generalized to carry observability and non-observability contexts. This
+includes SpanContext, Distributed-context, Authentication-context, etc. Distributed-context
+carries key-value pairs for observability. The name Distributed-context is
+very confusing. Hence it should be changed to TagMap (or other better suggestion). 
+That is the first motivation.
+ 
+The second motivation is the restrict use of Distributed-context to observability. As it is
+written currently it does not specifically call out for using it only for non-observability purpose.
+However, it can be used to support 'Baggage' functionality for non-observability purpose as it is
+OpenTracing.
+There are few reasons to have different mechanism to address observability and
+non-observability use cases.
+1. For observability, key-value pairs SHOULD be write-only from application perspective. SDKs can read them
```

Comment:
Is write-only desirable? I figured distributedcontext entries would be something I could pipe into metrics as I see fit.

For example, if I used this to propagate my originating service, There's cases I would want metrics to also have the originating service, as I may want to split my metric by that dimension.

Reply:
SDK can read it so it can be applied to metrics.

Reply:
When you say the sdk, are you referring to the exporters?
In the case of metrics, I think a lot of the dimensions will be specified and passed in during the invocation (e.g. during the creation of a TimeSeries (currently terminology) or Handler (new RFC-based terminology)). In that world, DistributedContext would need to be readable by the user:

```
# create the main counter
request_count = Counter()

# in the request, get the relevant time series and add the value.
time_series = request_count.get_time_series({"service": DistributedContextSingleton.get("service"))
time_series.add(1)
```

Although it is possible for the exporter to have access to the values, there's no easy mechanism to tell the exporter which labels matter. 

Also this behavior is also valuable because metrics will pre-aggregate, so they need values to determine the aggregation before the measurement even reaches the exporter.